spark/src/main/scala/geotrellis/spark/reproject/TileRDDReproject.scala

```diff
@@ -269,4 +283,88 @@ object TileRDDReproject {
       apply(fakeBuffers, rdd.metadata, destCrs, targetLayout, options)
     } else
       apply(rdd.bufferTiles(bufferSize), rdd.metadata, destCrs, targetLayout, options)
+
+
+  /** Match pixel resolution between two layouts in different projections such that
+    * cell size in target projection and layout matches the most resolute tile.
+    *
+    * Note: that the ammount of pixel resolution distortion during the reprojection
+    * depends on the location of the extent being reprojeected.
+    */
+  private
+  def matchReprojectRasterExtent(
+    src: CRS, dst: CRS,
+    layout: LayoutDefinition,
+    keyBounds: Option[KeyBounds[SpatialKey]]
+  )(implicit sc: SparkContext): ReprojectSummary = {
+    // Bounds of tiles we need to examine
+    val bounds: GridBounds = keyBounds match {
+      case Some(kb) =>
+        kb.toGridBounds
+      case None =>
+        GridBounds(0, 0, layout.layoutCols, layout.layoutRows)
+    }
+
+    val mapTransform = layout.mapTransform
+    val getRasterExtent: (Int, Int) => RasterExtent = { (col, row) =>
+      RasterExtent(mapTransform(col, row), layout.tileCols, layout.tileRows)
+    }
+
+    val chunks = bounds.split(512, 512).toVector
+    sc.parallelize(chunks, chunks.length)
+      .map { boundsChunk =>
+        import scala.concurrent._
+        import scala.concurrent.duration._
+        import ExecutionContext.Implicits.global
+
+        val splitWork: Iterator[Future[ReprojectSummary]] =
+          boundsChunk.split(128,128) // large enough to be worth a Future
+            .map { subChunk =>
+              Future {
+                // Proj4Transform is not thread safe
+                val transform = Proj4Transform(src, dst)
```

Comment:
Should we add here a `synchronized` call?

Reply:
Btw i know that it's still WIP, but probably it makes sense to make execution context somehow configurable.

Reply:
My thought was that `synchronized` is just going to introduce resource contention and defeat the point of forking a `Future` to do the reprojection.

What is your concern about the choice of executor? This is CPU bound computation and the default executor is supposed well suited for those.

Reply:
Hm and what did you mean by thread safety of `Proj4Transform`?

In terms of limited sources it makes sense to have at least configurable pool, to control parallelizm level. But I'm fine with the default pool.